unlock-js/src/__tests__/unlockService.test.js

```diff
@@ -1,204 +1,108 @@
-          unlockAddress
-        )
-        expect(
-          unlockService._getImplementationAddressFromProxy
-        ).toHaveBeenCalledWith(unlockAddress)
-      })
-
-      it('should throw an error of the opCode is not a proxy', async () => {
-        expect.assertions(3)
-        unlockService._contractAbiVersionFromAddress = jest.fn(() => {
-          return
-        })
-        unlockService._getImplementationAddressFromProxy = jest.fn(() => {
-          throw new Error('not a proxy')
-        })
-        try {
-          await unlockService.unlockContractAbiVersion()
-        } catch (error) {
-          expect(error.message).toEqual(Errors.UNKNOWN_CONTRACT)
-        }
-        expect(
-          unlockService._contractAbiVersionFromAddress
-        ).toHaveBeenCalledWith(unlockAddress)
-        expect(
-          unlockService._getImplementationAddressFromProxy
-        ).toHaveBeenCalledWith(unlockAddress)
-      })
-
-      // TODO: this test should be removed and replaced by the next skipped test
-      it('should default to v0 if the version still could not be identified by the proxy', async () => {
-        expect.assertions(3)
-        const implementationAddress = '0x123'
-        const proxyOpCode = '0xabc'
-        const implementationOpCode = '0xdef'
-        unlockService.web3.eth = {
-          getCode: jest.fn(address => {
-            if (address == implementationAddress) {
-              return Promise.resolve(implementationOpCode)
-            } else {
-              // Proxy!
-              return Promise.resolve(proxyOpCode)
-            }
-          }),
-        }
-
-        unlockService._getImplementationAddressFromProxy = jest.fn(() => {
-          return implementationAddress
-        })
-
-        const version = await unlockService.unlockContractAbiVersion()
-        expect(version).toEqual(v0)
-        expect(unlockService.web3.eth.getCode).toHaveBeenCalledWith(
-          unlockAddress
-        )
-        expect(
-          unlockService._getImplementationAddressFromProxy
-        ).toHaveBeenCalledWith(unlockAddress)
-      })
-
-      // TODO: restore that when we have the default behavior in place
-      it.skip('should throw UNKNOWN_CONTRACT if the opCode does not match any version', async () => {
-        expect.assertions(2)
-        unlockService.web3.eth = {
-          getCode: jest.fn(() => {
-            return Promise.resolve('0xabc')
-          }),
-        }
-        try {
-          await unlockService.unlockContractAbiVersion()
-        } catch (error) {
-          expect(error.message).toEqual(Errors.UNKNOWN_CONTRACT)
-        }
-        expect(unlockService.web3.eth.getCode).toHaveBeenCalledWith(
-          unlockAddress
-        )
-      })
+  describe('_getVersionFromContract', () => {
+    it('should return the version number if there is one', async () => {
+      expect.assertions(1)
+
+      nock.ethCallAndYield(
+        '0x4220bd46',
```

Comment:
magic number.  maybe move to a constant or add a comment like // bytes4(keccak256('unlockVersion()'))

Reply:
Not sure where these deps are coming from? I added as comments.